src/react/databrowser/objects/ObjectList.jsx

```diff
@@ -74,11 +77,11 @@ export default function ObjectList({ objects, bucketNameParam, prefixParam }: Pr
         disableSortRemove: true,
         autoResetFilters: false,
         autoResetSortBy: false,
-    }, useFilters, useSortBy);
+    }, useFilters, useSortBy, useFlexLayout);
 
     return <L.ListSection>
         <T.ButtonContainer>
-            <T.ExtraButton icon={<i className="fas fa-upload" />} text="Upload" variant='info' onClick={() => dispatch(push(`/buckets/${bucketNameParam}/upload-object`))} size="default" type="submit" />
+            <T.ExtraButton icon={<i className="fas fa-upload" />} text="Upload" variant='info' onClick={() => dispatch(openObjectUploadModal())} size="default" type="submit" />
```

Comment:
Just a personal preference here really but I find if you split your attributes on new lines it makes reviewing side by side easier and viewing partial diffs simpler as it would highlight the specific change and not the entire element. Following a line length rule anyway, this helps. ( Roughly 100 characters per line is my max )

Example:
```suggestion
            <T.ExtraButton
                icon={<i className="fas fa-upload" />}
                text="Upload"
                variant='info'
                onClick={() => dispatch(openObjectUploadModal())}
                size="default"
                type="submit"
            />
```